trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,314 @@
+            if e.error_name in (
+                    error.NOT_FOUND,
+                    error.COLUMN_NOT_FOUND,
+                    error.TABLE_NOT_FOUND,
+            ):
+                return dict(text=None)
+            raise
+
+    def has_schema(self, connection: Connection, schema: str) -> bool:
+        query = f"SHOW SCHEMAS LIKE '{schema}'"
+        try:
+            res = connection.execute(sql.text(query))
+            return res.first() is not None
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.TABLE_NOT_FOUND,
+                    error.SCHEMA_NOT_FOUND,
+                    error.CATALOG_NOT_FOUND,
+            ):
+                return False
+            raise
+
+    def has_table(self, connection: Connection,
+                  table_name: str, schema: str = None, **kw) -> bool:
+        query = 'SHOW TABLES'
+        if schema:
+            query = f'{query} FROM {self.identifier_preparer.quote_identifier(schema)}'
+        query = f"{query} LIKE '{table_name}'"
+        try:
+            res = connection.execute(sql.text(query))
+            return res.first() is not None
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.TABLE_NOT_FOUND,
+                    error.SCHEMA_NOT_FOUND,
+                    error.CATALOG_NOT_FOUND,
+                    error.MISSING_SCHEMA_NAME,
+            ):
+                return False
+            raise
+
+    def has_sequence(self, connection: Connection,
+                     sequence_name: str, schema: str = None, **kw) -> bool:
+        """Trino has no support for sequence. Returns False indicate that given sequence does not exists."""
+        return False
+
+    def _get_server_version_info(self, connection: Connection) -> Tuple[int, ...]:
+        query = 'SELECT version()'
+        res = connection.execute(sql.text(query))
+        version = res.scalar()
+        return tuple([version])
+
+    def _get_default_schema_name(self, connection: Connection) -> Optional[str]:
+        dbapi_connection: trino_dbapi.Connection = connection.connection
+        return dbapi_connection.schema
+
+    def do_execute(self, cursor: Cursor, statement: str, parameters: Tuple[Any, ...],
+                   context: DefaultExecutionContext = None):
+        cursor.execute(statement, parameters)
+        if context and context.should_autocommit:
+            # SQL statement only submitted to Trino server when cursor.fetch*() is called.
+            # For DDL (CREATE/ALTER/DROP) and DML (INSERT/UPDATE/DELETE) statement, call cursor.description
+            # to force submit statement immediately.
+            cursor.description  # noqa
+
+    def do_rollback(self, dbapi_connection: trino_dbapi.Connection):
+        if dbapi_connection.transaction is not None:
+            dbapi_connection.rollback()
+
+    def do_begin_twophase(self, connection: Connection, xid: str):
+        pass
```

Comment:
Should these `pass` or `raise` something? We don't want an application to try to call `do_rollback_twophase` and then notice that the operation didn't get rolled back.

(Reminder to self - see what other clients do).

Reply:
Other clients (which don't support 2PC transactions) `raise NotImplementedError()` from this method (and all twophase methods).
Please change to `raise NotImplementedError()`.

Reply:
@hashhar Yeah, I just remove all twophase methods to leave it to default (`raise NotImplementedError()`)